### DISCUSSION TITLE
Hovered Tooltip

### DISCUSSION THREAD

**Discussion by Froxcey** [neoclide/coc.nvim#3984] [Q&A] [answered]
I write Typescript and I'm starting to write Rust code. I have installed both coc-tsserver and coc-rust-analyzer. Everything is perfectly fine when writing code, but I want the hover tooltip when reading other people's code. Is there anyway I can hover something in normal mode using the vim cursor and use a keybind to show documentation tooltip?

**Comment by fannheyward** [comment, marked as answer]
```
" Use K to show documentation in preview window.
nnoremap <silent> K :call ShowDocumentation()<CR>

function! ShowDocumentation()
  if CocAction('hasProvider', 'hover')
    call CocActionAsync('doHover')
  else
    call feedkeys('K', 'in')
  endif
endfunction
```

**Reply by Froxcey** [reply]
Thanks! That works perfectly

**Comment by chemzqm** [comment]
You can use `<MouseMove>` key-mapping on latest gvim >= 8.2.4674, see `:h <MouseMove>`

**Reply by vitaly-zdanevich** [reply]
Not in Vim?

**Reply by chemzqm** [reply]
It's gvim only, check the doc.
